custom_components/tahoma/__init__.py

```diff
@@ -40,17 +44,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         _LOGGER.exception(exception)
         return False
 
-    devices = await client.get_devices()
     scenes = await client.get_scenarios()
 
+    coordinator = TahomaDataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="Tahoma fetcher",
+        client=client,
+        devices=await client.get_devices(),
+        listener_id=await client.register_event_listener(),
+        update_interval=timedelta(seconds=5),
+    )
+
+    await coordinator.async_refresh()
+
     hass.data[DOMAIN][entry.entry_id] = {
         "controller": client,
```

Comment:
Could be removed? Looks not necessary anymore :).

Reply:
You are right.